src/wildcard-service/init.lua

```diff
@@ -89,7 +101,25 @@ local function resolve(service)
 end
 
 function _M:servers(host)
-  local domain_info = self:domain_info(host or _M.arg_host())
+  local domain = host or _M.arg_host()
+
+  local cached_domain_info, stale_domain_info = self.domain_cache:get(domain)
+
+  local domain_info = cached_domain_info or
+      stale_domain_info or
+      get_domain_info(false, domain, self.api_host, self.http_client, self.domain_cache, self.domain_cache_ttl)
+
+  -- When the cache info is stale, we use it, but also start a timer to refresh
+  -- it. This avoids blocking the request.
+  if stale_domain_info then
+    local ok, err = ngx.timer.at(
+      0, get_domain_info, domain, self.api_host, self.http_client, self.domain_cache, self.domain_cache_ttl)
```

Comment:
Why not to pass just `self` ?

Reply:
At first, I thought it would be a bit strange because `self` would not be the first param (it needs to be `premature` coming from the timer). However, know that you mention it, I'd rather pass `self`. I prefer to avoid passing so many params. Changed.